src/main/webapp/feed.js

```diff
@@ -0,0 +1,468 @@
+let mockFoodSteps = ["Food step 1", "Food step 2", "Food step 3"];
+let mockFoodStepDescriptions = ["food description 1",
+"food description 2", "food description 3"];
+let mockFoodStepResources = ["Here is a link for more information",
+"Here is a link for more information", "Here is a link for more information"];
+mockChallenges[2].set("steps", mockFoodSteps);
+mockChallenges[2].set("descriptions", mockFoodStepDescriptions);
+mockChallenges[2].set("resources", mockFoodStepResources);
+
+let mockUser = new Map();
+mockUser.set("name", "User A");
+mockUser.set("currentChallengeId", 1);
+// array where indices correspond to challenge id and value = how many steps have been completed
+mockUser.set("challengeStatuses", [1, 0, 2]);
+
+const user = mockUser;
+const challenges = mockChallenges;
+
+let eventCategoryIcons = new Map();
+eventCategoryIcons.set("food/drink", "🥑🍋🍏");
+eventCategoryIcons.set("nature", "🌲🌱🌳");
+
+function loadPage() {
+    const feed = document.getElementById("events-feed");
+    feed.appendChild(postEvent(mockEvent));
+    feed.appendChild(postEvent(mockEvent2));
+
+    setChallengeBox(user.get("currentChallengeId"));
+
+    setChallengesNavBar(mockUser, challenges);
+
+    window.onclick = function(event) {
+        const modal = document.getElementById("challenges-modal");
+        if (event.target == modal) {
+            closeChallengesModal();
+        }
+    }
+}
+
+function postEvent(event) {
+    const eventEl = document.createElement('div');
+    eventEl.className = "event-post";
+    eventEl.appendChild(addEventUserText(event));
+    eventEl.appendChild(addEventBookmark(event));
+    eventEl.appendChild(addEventMiddleSection(event));
+    eventEl.appendChild(addEventInfo(event));
+    return eventEl;
+}
+
+function addEventBookmark(event) {
+    const bookmarkDiv = document.createElement('div');
+    bookmarkDiv.style.height = 0;
+    const bookmark = document.createElement('img');
+    bookmark.className = "event-bookmark";
+    // add case for if user has bookmarked this event once users have been created
+    bookmark.src = "../resources/bookmark.png";
+    bookmarkDiv.appendChild(bookmark);
+    bookmarkDiv.appendChild(addEventNumBookmarks(event));
+    return bookmarkDiv;
+}
+
+function addEventNumBookmarks(event) {
+    const bookmarkNum = document.createElement('p');
+    bookmarkNum.className = "event-bookmark-num";
+    bookmarkNum.innerText = event.get("bookmarks");
+    return bookmarkNum;
+}
+
+function addEventUserText(event) {
+    const eventUser = document.createElement('p');
+    eventUser.className = "event-info";
+    eventUser.innerText = event.get("author") + " posted an event:";
+    return eventUser;
+}
+
+function addEventMiddleSection(event) {
+    const eventMiddle = document.createElement('div');
+    eventMiddle.className = "event-middle";
+    eventMiddle.appendChild(addEventInnerCard(event));
+    eventMiddle.appendChild(addEventDescription(event));
+    return eventMiddle;
+}
```

Comment:
Could this be abstracted? It feels like the code is pretty repetitive between these functions